src/tools/rbd_wnbd/rbd_wnbd.cc

```diff
@@ -719,6 +742,13 @@ class RBDService : public ServiceBase {
     int image_map_timeout;
     bool remap_failure_fatal;
 
+    std::thread adapter_monitor_thread;
+
+    ceph::mutex start_lock = ceph::make_mutex("RBDService::StartLocker");
+    ceph::mutex shutdown_lock = ceph::make_mutex("RBDService::ShutdownLocker");
```

Comment:
Not sure I understand the purpose of these locks as they are taken only in `run_hook` and `stop_hook` respectively.  Can we really end up with two concurrent e.g. `run_hook` invocations under the Windows service manager?

Reply:
The documentation doesn't seem to explicitly guarantee it so I'd rather stay cautious and use a lock.

https://learn.microsoft.com/en-us/windows/win32/api/winsvc/nc-winsvc-lpservice_main_functiona